Gems/AWSCore/Code/Include/Framework/AWSApiRequestJob.h

```diff
@@ -277,22 +277,30 @@ namespace AWSCore
         virtual void DoCleanup()
         {
         }
+    };
 
+    template<class RequestTraits>
+    class AwsApiRequestJob : public AwsApiRequestJobBase<RequestTraits>
```

Comment:
Curious: Can we use the AwsApiRequestJobType alias here instead of AwsApiRequestJobBase? I think AwsApiRequestJobType should be complete since it's defined in AwsApiRequestJobBase now.

Reply:
We can't because of the template parameter in this case.